File: data/EXP01/golang_no_system_var/2671/2671.output_1.go
type FrequencyTracker struct {
    numbers map[int]int
}

func Constructor() FrequencyTracker {
    return FrequencyTracker{
        numbers: make(map[int]int),
    }
}

func (ft *FrequencyTracker) Add(number int) {
    ft.numbers[number]++
}

func (ft *FrequencyTracker) DeleteOne(number int) {
    if _, ok := ft.numbers[number]; ok {
        ft.numbers[number]--
        if ft.numbers[number] == 0 {
            delete(ft.numbers, number)
        }
    }
}

func (ft *FrequencyTracker) HasFrequency(frequency int) bool {
    for _, count := range ft.numbers {
        if count == frequency {
            return true
        }
    }
    return false
}

File: data/EXP01/python/2671/2671.output_1.py
# @lc app=leetcode id=2671 lang=python3
class FrequencyTracker:
    def __init__(self):
        self.frequency_dict = {}

    def add(self, number):
        if number in self.frequency_dict:
            self.frequency_dict[number] += 1
        else:
            self.frequency_dict[number] = 1

    def deleteOne(self, number):
        if number in self.frequency_dict:
            self.frequency_dict[number] -= 1
            if self.frequency_dict[number] == 0:
                del self.frequency_dict[number]

    def hasFrequency(self, frequency):
        return frequency in self.frequency_dict.values()